packages/app/src/cli/models/app/app.ts

```diff
@@ -20,13 +25,85 @@ export const LegacyAppSchema = zod
   .strict()
 
 // adding http or https presence and absence of new lines to url validation
-const validateUrl = (zodType: zod.ZodString) => {
+const validateUrl = (zodType: zod.ZodString, {httpsOnly = false, message = 'Invalid url'} = {}) => {
+  const regex = httpsOnly ? /^(https:\/\/)/ : /^(https?:\/\/)/
   return zodType
     .url()
-    .refine((value) => Boolean(value.match(/^(https?:\/\/)/)), {message: 'Invalid url'})
-    .refine((value) => !value.includes('\n'), {message: 'Invalid url'})
+    .refine((value) => Boolean(value.match(regex)), {message})
+    .refine((value) => !value.includes('\n'), {message})
 }
 
+const ensurePathStartsWithSlash = (arg: unknown) => (typeof arg === 'string' && !arg.startsWith('/') ? `/${arg}` : arg)
+const ensureHttpsOnlyUrl = validateUrl(zod.string(), {
+  httpsOnly: true,
+  message: 'Only https urls are allowed',
+}).refine((url) => !url.endsWith('/'), {message: 'URL can’t end with a forward slash'})
+
+const SubscriptionEndpointUrlValidation = ensureHttpsOnlyUrl.optional()
+const PubSubProjectValidation = zod.string().optional()
+const PubSubTopicValidation = zod.string().optional()
+// example Eventbridge ARN - arn:aws:events:us-west-2::event-source/aws.partner/shopify.com/1234567890/webhooks_path
+const ArnValidation = zod
+  .string()
+  .regex(
+    /^arn:aws:events:(?<aws_region>[a-z]{2}-[a-z]+-[0-9]+)::event-source\/aws\.partner\/shopify\.com(\.test)?\/(?<api_client_id>\d+)\/(?<event_source_name>.+)$/,
+  )
+  .optional()
+
+export const WebhookSubscriptionSchema = zod.object({
+  topic: zod.string(),
+  sub_topic: zod.string().optional(),
+  format: zod.enum(['json', 'xml']).optional(),
+  include_fields: zod.array(zod.string()).optional(),
+  metafield_namespaces: zod.array(zod.string()).optional(),
+  subscription_endpoint_url: SubscriptionEndpointUrlValidation,
+  path: zod
+    .string()
+    .refine((path) => path.startsWith('/') && path.length > 1, {
+      message: 'Path must start with a forward slash and be longer than 1 character',
+    })
+    .optional(),
+  pubsub_project: PubSubProjectValidation,
+  pubsub_topic: PubSubTopicValidation,
+  arn: ArnValidation,
```

Comment:
I was going to float the question about this, it would definitely simplify things! I think the biggest question was around the pub sub config. @karenxie @jenstanicak  what do we think?

Reply:
We call them endpoint types in the docs, could that work? 
<img width="870" alt="image" src="https://github.com/Shopify/cli/assets/61324669/15c6f898-1e61-4869-9eda-db26dab3078a">

Reply:
Well, the endpoint types are those 3 different options. But here we want to refer to the URI of each one where the event is going to be delivered:
- URL for HTTPS
- ARN for EventBridge
- Project::Topic for Pub/Sub

My suggestion was to use the same field name for all of them, to simplify the TOML structure, because we won't allow multiple endpoint types at the same time. Something like `callback_uri`, `destination_uri` or `endpoint_uri`.

> **Note**
> As @alexanderMontague commented recently, to be precise, we should use URI instead of URL. [URI](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier) is a general resource identifier, while [URL](https://en.wikipedia.org/wiki/URL) is a specific type of URI for Web (the ones starting with http, unlike the ARNs).

Reply:
Regarding the name, let me add some extra confusion:
- In the [REST API](https://shopify.dev/docs/api/admin-rest/2023-01/resources/webhook#resource-object) we call it `address`: _Destination URI to which the webhook subscription should send the POST request when an event occurs._ 
- In the [GraphQL API](https://shopify.dev/docs/api/admin-graphql/2023-01/objects/WebhookSubscription#fields) there's a property called `endpoint`, which can contain: 
  - `callbackUrl`
  - `arn`
  - `pubSubProject`/`pubSubTopic`
- In the [database](https://github.com/Shopify/shopify/blob/3e22840c399d6153c9ec1d8a59a897a4567bcbbd/db/shard_structure.versioned.sql#L22368) we store it as `address`.

So if we want to be consistent with the current APIs, maybe the most accurate term would be `address`, which matches what we have here (the URI for any kind of endpoint type). What do you think?